Add prog-face-refine

ideasman42 opened the pull request:
### Brief summary of what the package does

Small package to refine faces for comment, string types.

### Direct link to the package repository

https://codeberg.org/ideasman42/emacs-prog-face-refine

NOTE: checked if there is a package to do this and it seems not: https://www.reddit.com/r/emacs/comments/14pyf8x/package_to_customize_comment_faces/

### Your association with the package

Maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this - will merge, but with some feedback:

I think a user would have trouble teasing our what this package does from the docstrings and Commentary, e.g.  "Refine faces for comments & strings."  I had to read your reddit thread to understand what this is.  Maybe something about "extending" or providing "additional options" for font-locking comments and strings could be added to your Commentary or summary line.  Possibly add the examples from your reddit thread.

Super minor:
- Setting `fill-column: 99` is (I suspect, but have no evidence) wider than most Emacsers typically use, and I imagine you're doing this to allow documentation strings to be wider?  If it's all the same to you I'd recommend using the default so that you're not producing packages with idiosyncratically wide documentation for users
- "Highlight block under the cursor." -> I think this is an out-of-place docstring
- "Return nil if RE is not a valid regexp." -> I think this should read "non-nil" if I'm reading correctly

ideasman42 commented:
Thanks for checking.
Re: `fill-column: 99`, I find 80 a bit restrictive (wraps too much)
as it happens checkdoc ignores the fill-column, so doc-strings still need to fit within 79 chars.

- Corrected out-of-place & incorrectly worded doc-strings.

Other changes:

- Faces which are lists are now accepted (allows faces to inherit from others).
- Updated Usage text.
- Added more examples.
